hc-tree-view.html

```diff
@@ -14,11 +18,42 @@
         display: block;
       }
     </style>
+
+    <hc-nested-list items="{{items}}">
+      <template>
+        <div class="item" data-index="{{index}}">
+          <paper-checkbox></paper-checkbox>
+          <iron-icon icon="icons:add-circle-outline"></iron-icon>
+          <span>[[item.label]]</span>
+        </div>
+      </template>
+    </hc-nested-list>
   </template>
 
   <script>
     Polymer({
-      is: 'hc-tree-view'
+      is: 'hc-tree-view',
+      properties: {
+        items: Array,
+      },
+      /**
+       * Takes an array of items then finds all references and changes their selected status in the nested list.
+       *
+       * @param {Object[]} items
+       */
+      selectItems(items) {
+        const parts = Object.assign({}, this._parts);
+
+        items.forEach(ref => {
+          const id = ref.id;
+          const parentId = this._getParentId(ref);
+          const item = parts[parentId].find(item => item.id === id);
+
+          if (item) item.selected = true;
```

Comment:
This is a catch to avoid setting a property on an undefined object, item must exist or javascript thrown an exception when calling `item.selected`, whether you're reading or writing to it.